### PR TITLE
Damage ext example

### PR DESCRIPTION
**Pull request by mgarg1**
This example demonstrates the use of the damage extension. This simple script creates 2 windows:
1.Changing Window - continuously changes color
2.Tracking Window  - displays the changed area in the Changing Window

The area notified by the damage extension notification will be painted in the second window.

### CODE REVIEW

File: Xlib/ext/damage.py [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+
+def damage_create(self, level):
+    did = self.display.allocate_resource_id()
+    DamageCreate(display=self.display,
+                 opcode=self.display.get_extension_major(extname),
+                 damage=did,
+                 drawable=self.id,
+                 level=level,
+                 )
+    print('damage create')
```
Comment (vasily-v-ryabov) [inline Xlib/ext/damage.py]: It looks like debug print. Please remove here and below.

File: examples/xdamage.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/python
+#
+# examples/xdamage.py -- demonstrate damage extension
+#
+#    Copyright (C) 2002 Peter Liljenberg <petli@ctrl-c.liu.se>
```
Comment (vasily-v-ryabov) [inline examples/xdamage.py]: I think copyright can be your own one for this example.

File: examples/xdamage.py [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+def blink(display, win, gc, cols):
+    while 1:
+        time.sleep(2)
+        print('Changing color', cols[0])
+        gc.change(foreground = cols[0])
+        cols = (cols[1], cols[0])
+        redraw(win, gc)
+        display.flush()
+
+def getImageFromWin(win,ptW,ptH,ptX=0,ptY=0):
```
Comment (vasily-v-ryabov) [inline examples/xdamage.py]: `python-xlib` generally follows PEP8 style except names from protocol. So it's better to adjust this and below functions to such a signature:
```python
def get_image_from_win(win, pt_w, pt_h, pt_x=0, pt_y=0):
```

File: examples/xdamage.py [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+    # Check for extension
+    if not disp.has_extension('DAMAGE'):
+        sys.stderr.write('server does not have the DAMAGE extension\n')
+        sys.stderr.write("\n".join(disp.list_extensions()))
+        
+        if disp.query_extension('DAMAGE') is None:
+            sys.exit(1)
+    else:
+        r = disp.damage_query_version()
+        print('DAMAGE version %d.%d' % (r.major_version, r.minor_version))
```
Comment (vasily-v-ryabov) [inline examples/xdamage.py]: `.format()` is recommended instead of `%` string interpolation.

File: examples/xdamage.py [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+        display.flush()
+
+def getImageFromWin(win,ptW,ptH,ptX=0,ptY=0):
+    try:
+        raw = win.get_image(ptX,ptY, ptW,ptH, X.ZPixmap, 0xffffffff)
+        image = Image.frombytes("RGB", (ptW, ptH), raw.data, "raw", "BGRX")
+        return image
+
+    except Exception:
+        print("exception thrown in getImageFromWin")
```
Comment (vasily-v-ryabov) [inline examples/xdamage.py]: This addition could be useful to understand what happened:
```python
import traceback
traceback.print_exc()
```

File: examples/xdamage.py [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+
+    while 1:
+        event = d.next_event()
+        if event.type == X.Expose:
+            if event.count == 0:
+                redraw(window1, gc)
+        elif event.type == d.extension_event.DamageNotify:
+            image = getImageFromWin(window1, event.area.width, event.area.height, event.area.x, event.area.y)
+            bgpm = window2.create_pixmap(image.width,image.height,d.screen().root_depth)
+            bggc = window2.create_gc(foreground=0,background=0)
```
Comment (vasily-v-ryabov) [inline examples/xdamage.py]: (minor) These 2 lines should contain spaces after every comma.

File: examples/xdamage.py [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+def get_image_from_win(win, pt_w, pt_h, pt_x=0, pt_y=0):
+    try:
+        raw = win.get_image(ptX,ptY, ptW,ptH, X.ZPixmap, 0xffffffff)
+        image = Image.frombytes("RGB", (ptW, ptH), raw.data, "raw", "BGRX")
+        return image
+
+    except Exception:
+        traceback.print_exc()
+
+def checkExt(disp):
```
Comment (vasily-v-ryabov) [inline examples/xdamage.py]: The same for `checkExt` -> `check_ext`.

File: examples/xdamage.py [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+            sys.exit(1)
+    else:
+        r = disp.damage_query_version()
+        print('DAMAGE version {}.{}'.format(r.major_version, r.minor_version))
+    
+def main():
+    d = display.Display()
+    root = d.screen().root
+
+    checkExt(d)
```
Comment (vasily-v-ryabov) [inline examples/xdamage.py]: The function call has not been changed.

File: Xlib/ext/damage.py (hunk 1 above)
Comment (vasily-v-ryabov) [reply Xlib/ext/damage.py]: All `print()` calls should be removed from `damage.py`. Only example code may print something.

**Comment by matanster** [comment]
If anyone has code that manages to get damage events from windows already present in the display (not ones created by the same code listening to events) that would be a nice addition!

**Pull request merged**